backend/typescript/services/implementations/animalTypeService.ts

```diff
@@ -0,0 +1,118 @@
+import PgAnimalType from "../../models/animalType.model";
+import {
+  IAnimalTypeService,
+  AnimalTypeRequestDTO,
+  AnimalTypeResponseDTO,
+} from "../interfaces/animalTypeService";
+import { getErrorMessage, NotFoundError } from "../../utilities/errorUtils";
+import logger from "../../utilities/logger";
+
+const Logger = logger(__filename);
+
+class AnimalTypeService implements IAnimalTypeService {
+  /* eslint-disable class-methods-use-this */
+  async getAnimalType(id: string): Promise<AnimalTypeResponseDTO> {
+    let animalType: PgAnimalType | null;
+    try {
+      animalType = await PgAnimalType.findByPk(id, { raw: true });
+      if (!animalType) {
+        throw new NotFoundError(`Animal type id ${id} not found`); // return json instead (for standardization of responses)?
+      }
+    } catch (error: unknown) {
+      Logger.error(`Failed to get entity. Reason = ${getErrorMessage(error)}`);
+      throw error;
+    }
+
+    return {
+      id: Number(animalType.id),
```

Comment:
Since animalType is defined with an id of type number in the database, when you retrieve an instance of animalType its id should already be a number. You shouldn't have to explicitly convert it
(I'm pretty sure)